ast/src/test/java/com/graphicsfuzz/common/typing/TyperTest.java

```diff
@@ -1000,6 +1000,56 @@ public void visitBinaryExpr(BinaryExpr binaryExpr) {
     }.visit(tu);
   }
 
+  @Test
+  public void testScalarSwizzleTyped() throws Exception {
+    final TranslationUnit tu = ParseHelper.parse("#version 420\n"
+        + "void main() {\n"
+        + "  float f;\n"
+        + "  vec3 fv;\n"
+        + "  int i;\n"
+        + "  ivec3 iv;\n"
+        + "  uint u;\n"
+        + "  uvec3 uv;\n"
+        + "  bool b;\n"
+        + "  bvec3 bv;\n"
+        + "  f.x;\n"
```

Comment:
Should also test stuff like `f.xxxx`, or `f.xx.xx` ?

Reply:
We perhaps could, but I actually only added this test to double check that GraphicsFuzz already did the right thing here (which it does) - I considered then removing the test, since I normally write tests for missing features, but thought I may as well leave it in. Thus I won't make it exhaustive.